mbf_abstract_nav/src/navigate_action.cpp

```diff
@@ -158,14 +158,40 @@ void NavigateAction::start(GoalHandle &goal_handle)
   }
 
   // start navigating with the split path
-  startNavigate();  
+  action_state_ = NAVIGATE;
+  startNavigate();
+
+  // double check if the plan request has 
+  if (action_state_ == SUCCEEDED)
+  {
+    geometry_msgs::PoseStamped robot_pose;
+    robot_info_.getRobotPose(robot_pose);
+    ROS_INFO("Succeeded from navigation, double checking for orientation from mbf");
+    if (mbf_utility::distance(robot_pose, plan.checkpoints.back().pose) <= plan.xy_goal_tolerance 
+      && mbf_utility::angle(robot_pose, plan.checkpoints.back().pose) <= plan.yaw_goal_tolerance)
+    {
+      ROS_INFO_STREAM_NAMED("navigate", "Plan complete with desired goal tolerance");
+      navigate_result.status = forklift_interfaces::NavigateResult::SUCCESS;
+      navigate_result.remarks = "Action navigate completed successfully!";
+      navigate_result.final_pose = robot_pose;
```

Comment:
Is it fine to not populate the distance to goal and angle to goal fields of the result?

Reply:
i see, will add it just to be sure. thanks